### PR TITLE
Hl7 12 po parse message 2 save raw message

### PR DESCRIPTION
**Pull request by Perry5**
- https://jira.amida.com/browse/HL7-12

- Adds new mongo schema
- implements basic file parsing (but not message parsing)
- adds `passport` for authenticating endpoints
- factors out mongo connection setup
- adds dropDatabase.js script that runs before running tests

- oh so many breaking changes
  - upload endpoint is authenticated now
  - entire mongo schema is changed

- Pester Grant to add these

- N/A

### CODE REVIEW

File: server/hl7/hl7.model.js [hunk 1]
```diff
@@ -1,8 +1,24 @@
...
 
-const HL7FileUploadSchema = new mongoose.Schema({
-  fileName: { type: String, required: true },
-  dateAdded: { type: Date, default: () => Date.now() },
+const MessageSchema = new mongoose.Schema({
+  fileId: {
+    type: mongoose.Schema.ObjectId,
+    required: true,
+  },
+  indexWithinFile: {
```
Comment (Perry5) [inline server/hl7/hl7.model.js]: Is this the position of a particular message within a file? So each message being on a separate line?

File: server/hl7/hl7.controller.js [hunk 2]
```diff
@@ -24,27 +29,61 @@ const upload = multer({ storage,
...
+    ADT: rawHl7Message
+  };
+}
+
+/**
+ * Reads a file from a specified path and returns a list of hl7 messages
+ * @param filePath
+ * @returns {*}
+ */
+function readFile(filePath) {
```
Comment (Perry5) [inline server/hl7/hl7.controller.js]: Do we still need this function if `parseFile()` handles reading of the file?

File: server/hl7/hl7.controller.js [hunk 3]
```diff
@@ -24,27 +29,61 @@ const upload = multer({ storage,
...
+    req.user.files.unshift({ filename: req.file.path });
+    const newFile = req.user.files[0];
+    return Promise.all([data, newFile, req.user.save()]);
+  }).then(([data, file]) => {
+    const hl7MessageList = data
+      .replace(/\n\r/g, '\n')
+      .replace(/\r/g, '\n')
+      .split(/\n{2,}/g);
 
+    return Message.create(hl7MessageList.filter(rawMessage => !!rawMessage)
```
Comment (Perry5) [inline server/hl7/hl7.controller.js:62]: It doesn't seem like we save meta data about the uploaded file like we were doing before. Is that something we still want to do?

File: server/hl7/hl7.model.js [hunk 4]
```diff
@@ -1,8 +1,24 @@
 const mongoose = require('mongoose');
 
-const HL7FileUploadSchema = new mongoose.Schema({
-  fileName: { type: String, required: true },
-  dateAdded: { type: Date, default: () => Date.now() },
+const MessageSchema = new mongoose.Schema({
+  fileId: {
```
Comment (Perry5) [inline server/hl7/hl7.model.js:4]: Where is this field supposed to be coming from?

Comment (Perry5) [reply server/hl7/hl7.model.js:4]: Wouldn't this just be a duplicate with the mongo `_id` field.

**Review by Perry5 — COMMENTED** [review]
- When I post/upload a new file, nothing gets written to the `hl7fileuploads` collection, just the `messages` collection.

File: server/hl7/hl7.model.js (hunk 1 above)
Comment (orndorffgrant) [reply server/hl7/hl7.model.js]: good q, the intention is: if `indexWithinFile` is `6` then this message was the 6th message in the file specified by `fileId`.
It is not the line within a file at which a message starts.
Naming suggestions to make that clearer are welcome :)

File: server/hl7/hl7.controller.js (hunk 2 above)
Comment (orndorffgrant) [reply server/hl7/hl7.controller.js]: ah I guess not, unless we think we'll need that functionality elsewhere, in which case we could factor it out

File: server/hl7/hl7.controller.js (hunk 3 above)
Comment (orndorffgrant) [reply server/hl7/hl7.controller.js:62]: it is now within the user's document!

File: server/hl7/hl7.model.js (hunk 4 above)
Comment (orndorffgrant) [reply server/hl7/hl7.model.js:4]: this comes from the subdocument within the user's document for the file that this message was parsed out of. e.g. `user.files[0]._id`

**Pull request merged**